backtracking/subset_sum.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file
+ * @brief Program to Count number of subsets
+ *
+ * @details Subset problem (https://en.wikipedia.org/wiki/Subset_sum_problem)
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief subset sum algorithm
+ */
+namespace backtracking {
+/**
+ * @namespace Subsets
+ * @brief Functions for counting subsets in a given array with a given sum
+ */
+namespace Subsets {
+/**
+ * @brief The main function implements count of subsets
+ * @param sum is the required sum of any subset
+ * @param in_arr is the input array
+ * @returns count of the number of subsets with required sum
+ */
+
+std::uint64_t subset_sum(int sum, const std::vector<int> &in_arr) {
+    int nelement = in_arr.size();  // number of subset element
+    int count_of_subset = 0;
+
+    for (int i = 0; i < (1 << (nelement)); i++) {
+        std::vector<int> subset;
+        for (int j = 0; j < nelement; j++) {
+            if (i & (1 << j)) {
+                subset.push_back(in_arr[j]);
+            }
+        }
+        int check = 0;
+        for (int k : subset) {
+            check += k;
+        }
+        if (check == sum) {
+            count_of_subset++;
+        }
+    }
+    return count_of_subset;
+}
+}  // namespace Subsets
+}  // namespace backtracking
+
+/**
+ * @brief Test implementations
+ * @returns void
+ */
+static void test() {
```

Comment:
An interesting corner case might be an empty array.

Reply:
💯 Added